src/core/choroplethFeature.js

```diff
@@ -0,0 +1,245 @@
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth = function (arg1, arg2) {
+    if (arg1 === undefined) {
+      return m_choropleth;
+    }
+    if (typeof arg1 === 'string' && arg2 === undefined) {
+      return m_choropleth[arg1];
+    }
+    if (arg2 === undefined) {
+      var choropleth = $.extend(
+        {},
+        m_choropleth,
+        arg1
+      );
+      m_choropleth = choropleth;
+    } else {
+      m_choropleth[arg1] = arg2; //if you pass in accessor for prop
+    }
+    m_this.modified();
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A uniform getter that always returns a function even for constant values.
+   * If undefined input, return all the choropleth values as an object.
+   *
+   * @param {string|undefined} key
+   * @return {function}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth.get = function (key) {
+    if (key === undefined) {
+      var all = {}, k;
+      for (k in m_choropleth) {
+        if (m_choropleth.hasOwnProperty(k)) {
+          all[k] = m_this.choropleth.get(k);
+        }
+      }
+      return all;
+    }
+    return geo.util.ensureFunction(m_choropleth[key]);
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds a polygon feature to the current layer.
+   *
+   * @param {array} coordinateArray
+   * @param {geo.color} fillColor
+   * @return {geo.feature}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._addPolygonFeature = function(coordinateArray, fillColor){
+    return m_this.layer()
+      .createFeature('polygon', {
+      })
+      .data([{
+        type: 'Polygon',
+        coordinates: coordinateArray
+      }])
+      .polygon(function (d) {
+        return {
+          'outer': d.coordinates[0],
+          'inner': d.coordinates[1] // undefined but ok
+        };                              
+      })
+      .position(function(d){
+        return {
+          x: d[0],
+          y: d[1]
+        };
+      })
+      .style({
+        'fillColor': fillColor
+      });
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds polygons from a given feature to the current layer.
+   *
+   * @param {} geoJsonFeature
+   * @param geo.color
+   * @return [{geo.feature}] 
+   */
+  ////////////////////////////////////////////////////////////////////////////          
+  this._featureToPolygons = function (feature, fillValue) {
+
+    var arrayOfPolygonFeatures = [];
+    
+    if (feature.geometry.type === 'Polygon'){
+      arrayOfPolygonFeatures = [
+        m_this
+          ._addPolygonFeature(feature.geometry.coordinates, fillValue)
+      ];
+      
+    }
+    if (feature.geometry.type === 'MultiPolygon') {
```

Comment:
@ronichoudhury if I am not mistaken, in protovis, we had Mark classes to represent more than one mark. Please see here: http://mbostock.github.io/protovis/jsdoc/symbols/pv.Mark.html

> I don't mean to belabor the point but... even in a GIS context, "feature"
> is in fact the wrong word to describe a collection of objects right?  Your
> GIS definition marks "feature class" for that meaning, and "feature" for
> the individual object.

feature in GIS context means that collection of entities that share same or similar data. Also, this is not new in graphic packages as well. Renaming pointFeature to pointFeatures would be possible but somehow that has a odd feeling to it. pointFeature and point or dot has two distinct meanings.